onnxruntime/core/providers/cpu/tensor/slice_helper.h

```diff
@@ -11,28 +11,40 @@ namespace onnxruntime {
 namespace SliceOp {
 // compute output_dims without steps (Slice V1-9 & DynamicSlice)
 // Please note this will not Flatten the output shape
-inline Status PrepareForComputeHelper(const std::vector<int64_t>& raw_starts,
-                                      const std::vector<int64_t>& raw_ends,
-                                      const std::vector<int64_t>& raw_axes,
+inline Status PrepareForComputeHelper(const gsl::span<const int64_t>& raw_starts,
+                                      const gsl::span<const int64_t>& raw_ends,
+                                      const gsl::span<const int64_t>& raw_axes,
                                       SliceOp::PrepareForComputeMetadata& compute_metadata) {
   // Initialize axes to the provided axes attribute or to the default sequence
-  std::vector<int64_t> axes(raw_axes);
-  if (axes.empty()) {
+  TensorShapeVector axes;
+  if (raw_axes.empty()) {
     //axes are omitted, they are set to[0, ..., ndim - 1]
-    axes.resize(raw_starts.size());
-    std::iota(axes.begin(), axes.end(), 0);
+    axes.reserve(raw_starts.size());
+    for (int64_t i = 0, limit = raw_starts.size(); i < limit; ++i) {
+      axes.push_back(i);
+    }
+  } else {
+    axes.reserve(raw_axes.size());
+    axes.assign(raw_axes.cbegin(), raw_axes.cend());
   }
 
   // Iterate through the provided axes and override the start/end ranges
-  std::unordered_set<int64_t> unique_axes;
-  const auto& dimension_count = compute_metadata.input_dimensions_.size();
-  for (size_t axis_index = 0, axes_count = axes.size(); axis_index < axes_count; ++axis_index) {
+  using AxesSet = pmr::InlinedHashSet<int64_t>;
+  const auto axes_count = axes.size();
+  const auto axes_buffer_size = EstimateInlinedHashSetMemory<int64_t>(axes_count);
```

Comment:
Calling EstimateInlinedHashSetMemory, OrtDeclareAllignedStackOrAllocatedBuffer and then using SmallBufferResource to get a hash set seems like a lot of detail spilling outside. Can this be encapsulated behind a factory function like CreateInlinedHashSet or CreateHashSet (in case we don't want to use the inlined ver) that takes the number of elements and returns the set?

Reply:
I could not figure out a way to do it. This is mostly due to alloca. This technique to preallocate memory on the stack if possible is only for performance critical scenarios.
Most of the time, we can simply use InlinedHashSet.

Reply:
Alloca has been removed. No need for all of this machinery.